packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -216,6 +229,78 @@ export class RunningSummarizer implements IDisposable {
 		});
 	}
 
+	public async handleSummaryAck(): Promise<number> {
+		const lastAck: IAckedSummary | undefined = this.summaryCollection.latestAck;
+		let refSequenceNumber = -1;
```

Comment:
As per my [other comment](https://github.com/microsoft/FluidFramework/pull/14142/files#r1106412267), this should not return -1. It should wait for ack and use its reference sequence number.

Reply:
THere might not be an ack. Isn't it ? We can't wait forever here. if it is available, great, if not proceed.

Reply:
I see. The very first time it runs, there may not be an ack so it should not wait.

My concern is that its assuming that this.summaryCollection.latestAck will be there when its called (after the first time it runs). The caller awaits the ack so this will be true today but it has the potential to go wrong if that code is modified.

I would rather pass in an ack to this function and it simply handles that ack. Then the two calll sites are:
1. During start - this.summaryCollection.lastAck is passed.
2. From handleSummaryAcks() - It will pass on the result of await this.summaryCollection.waitSummaryAck(refSequenceNumber).

Reply:
Not sure I follow -  it does not expect the latestAck to be there at all.  The idea is that only handleSummaryAcks will wait for the summary ack,

Reply:
It does not expect it directly. But there are 2 issues that can happen:
1. If await latestAck is not called before calling this, it will unncessarily call refresh on container runtime which would end up in a no-op. The (unsaid) expected usage is that the caller makes sure latest ack is awaited. I am suggesting making that expectation explicit.
2. If there is no ack, it will return -1. The caller has to handle this scenario. handleSummaryAcks() does not handle this scenario because it knows there is an ack before it calls this function. But its not explicitly spelled out and so there are chances to the caller making mistakes.

I believe we are saying the same thing :-). I am just suggesting making things more explicit so that failures don't go unnoticed causing other issues.

Reply:
At the very least, handleSummaryAcks() should assert that the return value from this function is not -1.

Reply:
I see. Added assert,